[CI/Build]: Centralize requirement files

hickeyma opened the pull request:
Move requirement files to one central directory `requirements` and remove `requirements` prefix from each file.

Update all files accordingly that access or reference requirement files.

The advantages of these changes are as follows:

- All requirements are in one location
- Removes duplication of files
- Minimizes errors and not needed dependencies

Partial #594

hickeyma committed:
Centralize requirement files

Move requirement files to one central directory `requirements` and
remove `requirements` prefix from each file.

Update all files accordingly that access or reference requirement files.

The advantages of these changes are as follows:

- All requirements are in one location
- Removes duplication of files
- Minimizes errors and unrequired dependencies

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma committed:
Update the docker build path

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

sammshen commented on .buildkite/pipeline.yml:
```diff
@@ -19,7 +19,7 @@ steps:
     plugins:
       - cache#v1.5.2:
-          key: "venv-{{ checksum \"requirements.txt\" }}-{{ checksum \"requirements-test.txt\" }}"
+          key: "venv-{{ checksum \"requirements/commonm.txt\" }}-{{ checksum \"requirements/test.txt\" }}"
```
misspelling here?

hickeyma replied:
Nice catch. Fixed.

sammshen commented:
LGTM (new folder look pretty neat to me)! However, I think l should leave this decision to those who have been working on the project much longer @Siddhant-Ray @ApostaC defer the final decision on this design to you guys

Siddhant-Ray reviewed (commented):
Looks great to me, however deferring final decision to @ApostaC as this may have long term effects.

hickeyma committed:
Update after review

Review comment:
- https://github.com/LMCache/LMCache/pull/695#pullrequestreview-2858688803

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

YaoJiayi reviewed (approved):
LGTM!
